### PR TITLE
plugin/ceph: increase the OSD metadata threshold warning

### LINKED ISSUE #434: OSD metadata warning needs revisit
```
ubuntu@juju-180f3e-15-lxd-0:~$ sudo ceph osd df tree
ID   CLASS  WEIGHT     REWEIGHT  SIZE     RAW USE  DATA     OMAP     META      AVAIL    %USE  VAR   PGS  STATUS  TYPE NAME           
 -1         654.96509         -  655 TiB  125 GiB   66 GiB  6.2 MiB    59 GiB  655 TiB  0.02  1.00    -          root default        
-12         218.32169         -  218 TiB   41 GiB   22 GiB  2.8 MiB    19 GiB  218 TiB  0.02  0.98    -              rack AZ1        
-11          43.66434         -   44 TiB  8.0 GiB  4.5 GiB  1.2 MiB   3.5 GiB   44 TiB  0.02  0.96    -                  host node-01
  2    ssd    7.27739   1.00000  7.3 TiB  1.4 GiB  909 MiB    2 KiB   538 MiB  7.3 TiB  0.02  1.02  101      up              osd.2   
  6    ssd    7.27739   1.00000  7.3 TiB  1.2 GiB  855 MiB  1.2 MiB   420 MiB  7.3 TiB  0.02  0.90  114      up              osd.6   
 13    ssd    7.27739   1.00000  7.3 TiB  1.8 GiB  814 MiB    1 KiB  1000 MiB  7.3 TiB  0.02  1.28  105      up              osd.13  
 21    ssd    7.27739   1.00000  7.3 TiB  1.1 GiB  670 MiB    1 KiB   497 MiB  7.3 TiB  0.02  0.82   96      up              osd.21  
 31    ssd    7.27739   1.00000  7.3 TiB  1.2 GiB  716 MiB    1 KiB   549 MiB  7.3 TiB  0.02  0.89   99      up              osd.31  
 43    ssd    7.27739   1.00000  7.3 TiB  1.2 GiB  669 MiB    3 KiB   556 MiB  7.3 TiB  0.02  0.86  103      up              osd.43  
-15          43.66434         -   44 TiB  8.2 GiB  4.5 GiB  1.5 MiB   3.7 GiB   44 TiB  0.02  0.99    -                  host node-04
  4    ssd    7.27739   1.00000  7.3 TiB  1.4 GiB  736 MiB  210 KiB   741 MiB  7.3 TiB  0.02  1.04  111      up              osd.4   
  9    ssd    7.27739   1.00000  7.3 TiB  1.3 GiB  799 MiB   40 KiB   539 MiB  7.3 TiB  0.02  0.94  119      up              osd.9   
 16    ssd    7.27739   1.00000  7.3 TiB  1.3 GiB  652 MiB  273 KiB   650 MiB  7.3 TiB  0.02  0.92  102      up              osd.16  
 25    ssd    7.27739   1.00000  7.3 TiB  1.5 GiB  781 MiB   61 KiB   725 MiB  7.3 TiB  0.02  1.06   95      up              osd.25  
 36    ssd    7.27739   1.00000  7.3 TiB  1.4 GiB  770 MiB  297 KiB   656 MiB  7.3 TiB  0.02  1.00  112      up              osd.36  
 48    ssd    7.27739   1.00000  7.3 TiB  1.4 GiB  888 MiB  661 KiB   505 MiB  7.3 TiB  0.02  0.98  103      up              osd.48  
-21          43.66434         -   44 TiB  8.1 GiB  4.2 GiB    6 KiB   3.9 GiB   44 TiB  0.02  0.97    -                  host node-07
 12    ssd    7.27739   1.00000  7.3 TiB  1.8 GiB  878 MiB    1 KiB   938 MiB  7.3 TiB  0.02  1.28  113      up              osd.12  
 20    ssd    7.27739   1.00000  7.3 TiB  1.2 GiB  839 MiB    1 KiB   381 MiB  7.3 TiB  0.02  0.86  109      up              osd.20  
 30    ssd    7.27739   1.00000  7.3 TiB  1.2 GiB  714 MiB    1 KiB   546 MiB  7.3 TiB  0.02  0.89  107      up              osd.30  
 42    ssd    7.27739   1.00000  7.3 TiB  1.2 GiB  589 MiB    1 KiB   612 MiB  7.3 TiB  0.02  0.85   94      up              osd.42  
 52    ssd    7.27739   1.00000  7.3 TiB  1.4 GiB  610 MiB    1 KiB   810 MiB  7.3 TiB  0.02  1.00   89      up              osd.52  
 61    ssd    7.27739   1.00000  7.3 TiB  1.3 GiB  684 MiB    1 KiB   693 MiB  7.3 TiB  0.02  0.97  100      up              osd.61
[...]
````

In fresh clusters, the majority of the disk use is for the metadata. In the example above, there's ~80% of the use is for metadata which obviously triggers the warning (over the current threshold of 2%). 

I am wondering if we need apply a further limit such as raw size of the metadata is over 2G or something similar.

### PR DESCRIPTION
**Pull request by pponnuvel**
Fixes #434.

Signed-off-by: Ponnuvel Palaniyappan <pponnuvel@gmail.com>

### CODE REVIEW

File: hotsos/core/plugins/storage/ceph.py [hunk 1]
```diff
@@ -526,6 +526,10 @@ def bluefs_oversized_metadata_osds(self):
         for device in self.osd_df_tree['nodes']:
             if device['id'] >= 0:
                 meta_kb = device['kb_used_meta']
+                # Util under under 2G are ignored (treated as non-issue).
+                # See #434 for relevant info.
+                if meta_kb < (2 * 1024 * 1024):
```
Comment (taodd) [inline hotsos/core/plugins/storage/ceph.py:531]: 2 G is too small to detect an oversized META usage case, see https://github.com/canonical/hotsos/issues/434#issuecomment-1279652013

Comment (pponnuvel) [reply hotsos/core/plugins/storage/ceph.py:531]: I think you've mistaken the 2G as the absolute hard limit. We already test whether the meta util is > 5% and warn on that.
Which produces a lot of false positives, necessitating this tweak.

Comment (taodd) [reply hotsos/core/plugins/storage/ceph.py:531]: Ah, it's just a low limit, that makes sense.

**Review by taodd — APPROVED** [review]

File: tests/unit/fake_data_root/storage/ceph-mon/sos_commands/ceph_mon/json_output/ceph_osd_df_tree_--format_json-pretty [hunk 2]
```diff
@@ -94,10 +94,10 @@
             "pool_weights": {},
             "reweight": 1,
             "kb": 104837120,
-            "kb_used": 1048984,
+            "kb_used": 2097558,
```
Comment (dosaboy) [inline tests/unit/fake_data_root/storage/ceph-mon/sos_commands/ceph_mon/json_output/ceph_osd_df_tree_--format_json-pretty:97]: I know its a pain but if there is anyway that your test could make these changes on a temporary basis that would be preferable. I prefer to keep the contents of the fakle_data_roots "real" if possible.

Comment (dosaboy) [reply tests/unit/fake_data_root/storage/ceph-mon/sos_commands/ceph_mon/json_output/ceph_osd_df_tree_--format_json-pretty:97]: Try this: https://pastebin.canonical.com/p/fxbCY48BZn/

Comment (pponnuvel) [reply tests/unit/fake_data_root/storage/ceph-mon/sos_commands/ceph_mon/json_output/ceph_osd_df_tree_--format_json-pretty]: Applied, thanks. Also removed the short 7 full examples from 'functional' test as it no longer works (due to 'fake_data_root' not containing the data to cause the expected failures).

**Pull request merged**